gitlab/resource_gitlab_deploy_key_enable.go

```diff
@@ -0,0 +1,127 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabDeployEnableKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabDeployKeyEnableCreate,
+		Read:   resourceGitlabDeployKeyEnableRead,
+		Delete: resourceGitlabDeployKeyEnableDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceGitlabDeployKeyEnableStateImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"key_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"title": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"can_push": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGitlabDeployKeyEnableCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	key_id, err := strconv.Atoi(d.Get("key_id").(string))
+
+	log.Printf("[DEBUG] enable gitlab deploy key %s/%d", project, key_id)
+
+	deployKey, _, err := client.DeployKeys.EnableDeployKey(project, key_id)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s:%d", project, deployKey.ID))
+
+	return resourceGitlabDeployKeyEnableRead(d, meta)
+}
+
+func resourceGitlabDeployKeyEnableRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	deployKeyID, err := strconv.Atoi(d.Get("key_id").(string))
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] read gitlab deploy key %s/%d", project, deployKeyID)
+
+	deployKey, response, err := client.DeployKeys.GetDeployKey(project, deployKeyID)
+
+	// HTTP 404 indicates that the key is not enabled on the project, maybe it was removed through the UI
+	if response.StatusCode == 404 {
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	d.Set("title", deployKey.Title)
+	d.Set("key_id", deployKey.ID)
+	d.Set("key", deployKey.Key)
+	d.Set("can_push", deployKey.CanPush)
+	d.Set("project", project)
+	return nil
+}
+
+func resourceGitlabDeployKeyEnableDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+	deployKeyID, err := strconv.Atoi(d.Get("key_id").(string))
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] Delete gitlab deploy key %s/%d", project, deployKeyID)
+
+	response, err := client.DeployKeys.DeleteDeployKey(project, deployKeyID)
+
+	// HTTP 204 is success with no body
+	if response.StatusCode == 204 {
```

Comment:
we should just do response.StatusCode/100 == 2

Reply:
will do